packages/node-core/src/indexer/blockDispatcher/worker-block-dispatcher.ts

```diff
@@ -190,9 +192,12 @@ export abstract class WorkerBlockDispatcher<DS, W extends Worker>
   }
 
   private async getNextWorkerIndex(): Promise<number> {
-    return Promise.all(this.workers.map((worker) => worker.getMemoryLeft())).then((memoryLeftValues) => {
-      return memoryLeftValues.indexOf(Math.max(...memoryLeftValues));
-    });
+    this.currentWorkerIndex = (this.currentWorkerIndex + 1) % this.workers.length;
```

Comment:
This looks very tricky, you changed a class variable in a `get` method, which only suppose to be readonly.